lddwrap/__init__.py

```diff
@@ -0,0 +1,228 @@
+        if proc_unused.returncode not in [0, 1]:
+            raise RuntimeError("Failed ldd-ing external libraries of {} with code {}:\nout:\n{}\n\nerr:\n{}".format(
+                path, proc.returncode, out_unused, err_unused))
+
+    dependencies = dependencies_parse_helper(
+        out=out, out_unused=out_unused, verbose=verbose, unused=unused)  # type: List[Dependency]
+
+    return dependencies
+
+
+def dependencies_parse_helper(out: str, out_unused: str, verbose: Optional[bool] = None,
+                              unused: Optional[bool] = None) -> List[Dependency]:
+    """
+    Help to parse the command line output.
+
+    :param out: command line output
+    :param out_unused: command line output for ldd --unused
+    :param verbose: verbose flag set
+    :param unused: unused flag set
+    """
+    if verbose:
+        print(out)
+
+    dependencies = []  # type: List[Dependency]
+    for line in [line for line in out.split('\n') if line.strip()]:
+        # end of output
+        if "Version information" not in line:
+            dep = parse_line(line=line)
+            if dep is not None:
+                dependencies.append(dep)
+        else:
+            break
+
+    if unused:
+        dependencies = set_unused(dependencies=dependencies, out_unused=out_unused)
+
+    return dependencies
+
+
+def set_unused(dependencies: List[Dependency], out_unused: str) -> List[Dependency]:
+    """
+    Set Dependency argument unused.
+
+    :param dependencies: List of dependencies
+    :param out_unused: output from command ldd --unused
+    :return: List of dependencies
+    """
+    unused_dependencies = []  # type: List[pathlib.Path]
+    for line in [line for line in out_unused.split('\n') if line.strip()]:
+        # skip first line because it's no dependency
+        if line != "Unused direct dependencies:":
+            unused_dependencies.append(pathlib.Path(line.strip()))
+
+    for unused_dep in unused_dependencies:
+        for dep in dependencies:
+            # pylint: disable=simplifiable-if-statement
+            if dep.path == unused_dep:
+                dep.unused = True
+            else:
+                dep.unused = False
+
+    return dependencies
+
+
+def version(env: Optional[Dict[str, str]] = None) -> List[int]:
+    """
+    Return version of ldd.
+
+    :param env: the environment to use
+    :return: version X.Y
+    """
+    proc = subprocess.Popen(
+        ["ldd", "--version"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=env)
+
+    out, err = proc.communicate()
+    if proc.returncode != 0:
+        raise RuntimeError("Failed retrieving ldd version with code {}:\nout:\n{}\n\nerr:\n{}".format(
+            proc.returncode, out, err))
+    print(out)
+
+    # version is last element of first line
+    line = out.split('\n', 1)[0]
+    parts = [part.strip() for part in line.split(' ')]
+    version_str = parts[-1]
```

Comment:
Same here. Please make sure that there is at least one element in `parts` and if not raise an exception